Feature/image compare

Umebo opened the pull request:
Ta funkcja do wypisania nazwy jest zjebana, ale nie chcialo mi sie juz myslec

Commit:
Porownywanie obrazow z pewnym marginesem

Commit:
Dopasowanie i porownanie obrazow

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
 
-def saving_country(country_input):
+
+def SavingCountry(country_input):
```
twoje preferencje ale w pythonie raczej robi sie nazwy funckji saving_country

Antoszku replied:
#sugestia nazwa funkcij tez moglaby byc w stylu add_to_db(country)

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
+    pprint.pprint(db.countries.find_one({'name': country_name}))
+
+
+flag = Image.open('/home/igor/Documents/Python/GitKraken/Wikipedia_project/flag2.png').convert('RGB')
```
wystarczy open('flag2.png')

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
+    return flag_name[::-1]
+
+
+for file in glob.glob('/home/igor/Documents/Python/GitKraken/Wikipedia_project/Flags/*.png'):
```
wystarczy 'Flags/*.png'

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
+
+for file in glob.glob('/home/igor/Documents/Python/GitKraken/Wikipedia_project/Flags/*.png'):
+    test_flag = Image.open(file).resize(flag.size).convert('RGB')
+    if imgcompare.is_equal(flag, test_flag, tolerance = 20.0):
```
nie powinienes porownywac w ten sposob bo moze falszywie wyjsc porownanie, lepiej porownywac wszystkie po kolei i sprawdac gdzie jest najbardziej podobne

Umebo replied:
Zmienione

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
 import wikipedia
 import pprint
+from PIL import Image
```
dobre importowanie 👍

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
     print('You have that one already!')
-    pprint.pprint(db.countries.find_one({'name': country_name}))
\ No newline at end of file
+    pprint.pprint(db.countries.find_one({'name': country_name}))
```
czemu 2 razy pprint?

Umebo replied:
To jest metoda klasy ktra nazywa sie tak samo.

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
+
+flag = Image.open('/home/igor/Documents/Python/GitKraken/Wikipedia_project/flag2.png').convert('RGB')
+
+def FlagName(path):
```
flag_name 
Dodatkowo nie do końca jest dobra nazwa tej funkcji. Mało mówi. get_flag_name (albo get_flag_name_from(path)) byłoby chyba lepsze ale to tylko moja sugestia

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,45 @@
+            flag_name = flag_name + letter
+        else:
+            break
+    return flag_name[::-1]
```
tu można lekko usprawnić
```
if letter != '/':
       flag_name = flag_name + letter
else:
       return flag_name[::-1]
```

generalnei nie ma sensu tu uzywac breaka

Umebo replied:
yup

Commit:
Prownanie wszystkich obrazow, wybranie najmniejszej roznicy. Zmienione na funkcje.

Commit:
Wyszukiwanie zdań z podanym tagiem

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,74 @@
+    for file in glob.glob('Flags/*.png'):
+        test_flag = Image.open(file).resize(input_flag.size).convert('RGB')
+        percentage = imgcompare.image_diff_percent(test_flag, input_flag)
+        if percentage < difference:
```
dalej źle 👎 ciągle może być fałszywy wynik

Umebo replied:
dlaczego?

Umebo replied:
No może być, ale i tak to mi zwraca najbardziej podobną flagę

Antoch replied:
zwróci ci OSTATNIA flage ktorej percentage jest mniejszy niz 30 zamiast flagi ktora ma najmniejszy percetnage

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,74 @@
+    for phrase in whole_text:
+        if phrase.find(tag) != -1:
+            phrase = str(phrase)
+            tagged_phrases.append(phrase)
```
nie mam pewności ale z tego co pamiętam to masz zwracać tylko jedno zdanie więc możesz tu zrobić returna (a tagged_phrases do wywalenia)

Umebo replied:
"należy zwrócić tylko te zdania w których znajduje
się słowo 'Sea' w formie listy elementów"

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,74 @@
+    whole_text = TextBlob(db.countries.find_one({'name': country_name})['Summary']).sentences
+    tagged_phrases = []
+    for phrase in whole_text:
+        if phrase.find(tag) != -1:
```
możesz wyjaśnić co tu się dzieje XD?

Umebo replied:
To jest taki moduł który dzieli tekst na zdania (tworzy listę ze zdaniami jako elementami). Potem dla każdego zdania sprawdzam czy jest w nim dane słowo i ew. zapisuję do nowej listy.

Antoch replied:
to rozumiem tylko czemu -1 XD

Antoszku commented on Program_1.0.py:
```diff
@@ -1,22 +1,74 @@
+
 country_input = input(' Select country: ')
 country_name = wikipedia.WikipediaPage(country_input).title
+tag = input(' Lookin\' for sth specific? ')
```
XDDDDDD 10/10

Commit:
Poprawienie porównania flag. Zmiana nazw zmiennych.

Antoch commented on Program_1.0.py:
```diff
@@ -28,27 +28,29 @@ def flag_name_from_path(path):
-        if phrase.find(tag) != -1:
-            phrase = str(phrase)
-            tagged_phrases.append(phrase)
+    for sentence in whole_text:
```
very good 👍

Antoch commented on Program_1.0.py:
```diff
@@ -72,3 +76,6 @@ def phrase_with_tag(country_name, tag):
         data = phrase_with_tag(country_name, tag)
         for phrase in data:
             print(phrase)
+elif country_name == '' and tag != '':
```
#pierdola ale nie potrzbujesz tego :D

Umebo replied:
Zostawiłem to, bo będzie jeszcze else. Znaczy w sumie chyba to wszystko będzie do zmiany, skoro ma to być przez zapytania HTTP.

Antoch replied:
ideeowo jak robisz PR to maja tu byc TYLKO rzeczy ktore dotycza danego zadania i zero niepotrzebnego kodu :P w tym wypadku to jest niepotrzebny kod bo nic nie robi ale to tylko pierdola takze mozesz mergowac

Antoch reviewed (approved):
Generalnie approve - możesz mergowac - ewentualnie usun ta jedna linijke

Antoch commented on Program_1.0.py:
```diff
@@ -1,22 +1,81 @@
+
+
+def phrase_with_tag(country, tagged_word):
+    whole_text = TextBlob(db.countries.find_one({'name': country})['Summary']).sentences
```
ewentualnie ten whole_text mozna by zmienic bo troche malo mowi

Antoch replied:
taka pierdola moze na kiedys ale fajnie jakbys sam napisal kod ktory robi to co TextBlob

Commit:
Zmiana nazw, wywalenie ostatniej linijki, zapętlenie
